lpc55_sign/src/signed_image.rs

```diff
@@ -24,366 +36,166 @@ fn get_pad(val: usize) -> usize {
-
-    // SHA of each Root Key. This needs to go into each image and _must_
-    // match the SHA programmed in the CMPA area!
-    let root_hashes = certs.iter().map(|c| {
-        let root_bytes = std::fs::read(prefix.join(&c.cert_paths[0])).unwrap();
-        let (_, root0) = x509_parser::parse_x509_certificate(&root_bytes).unwrap();
-        let root0_pubkey =
-            rsa::RsaPublicKey::from_pkcs1_der(root0.public_key().subject_public_key.as_ref())
-                .unwrap();
-
-        // We need the sha256 of the pubkeys. This is just the sha256
-        // of n + e from the pubkey
-        let n = root0_pubkey.n();
-        let e = root0_pubkey.e();
-
-        let mut sha = sha2::Sha256::new();
-        sha.update(&n.to_bytes_be());
-        sha.update(&e.to_bytes_be());
-        let result = sha.finalize();
-        let mut v: Vec<u8> = vec![0; 32];
-        v.copy_from_slice(&result);
-        v
-    });
-
-    // We're relying on packed_struct to catch errors of padding
-    // or size since we know how big this should be
-    let cert_header_size = CertHeader::packed_bytes_size(None)?;
-
-    let mut new_cert_header: CertHeader = CertHeader::new(cert_header_size, cert_len);
-
-    new_cert_header.certificate_count = signing_chain.cert_paths.len() as u32;
-
-    // some math on how many bytes we sign
-    //
-    // Base image + padding
-    // certificate header block
-    // certificate length
-    // 4 sha256 hashes
-    let signed_len = bytes.len()
-        + image_pad
-        + cert_header_size
-        + (new_cert_header.certificate_table_len as usize)
-        + 32 * 4;
-
-    // Total image length includes 256 bytes of signature
-    let total_len = signed_len + 256;
-
-    new_cert_header.total_image_len = signed_len.try_into().unwrap();
-
-    let image_len = bytes.len();
-
-    byteorder::LittleEndian::write_u32(&mut bytes[0x20..0x24], total_len as u32);
-
-    let boot_field = BootField::new(BootImageType::SignedImage);
-
-    bytes[0x24..0x28].clone_from_slice(&boot_field.pack()?);
-    // Our execution address goes in the next word
-    byteorder::LittleEndian::write_u32(&mut bytes[0x34..0x38], execution_address);
-    // where to find the block. For now just stick it right after the image
-    byteorder::LittleEndian::write_u32(&mut bytes[0x28..0x2c], (image_len + image_pad) as u32);
-
-    let mut out = OpenOptions::new()
-        .write(true)
-        .truncate(true)
-        .append(false)
-        .create(true)
-        .open(outfile_path)?;
-
-    // Generate the image, see 7.3.4 of v2.4 UM 11126 for the layout
-    out.write_all(&bytes)?;
-    if image_pad > 0 {
-        out.write_all(&vec![0; image_pad])?;
-    }
-    out.write_all(&new_cert_header.pack()?)?;
-    out.write_all(&signing_certs)?;
-    let mut rkth_sha = sha2::Sha256::new();
-    for c in root_hashes {
-        out.write_all(&c)?;
-        rkth_sha.update(&c);
+) -> Result<Vec<u8>> {
+    if signing_certs.len() < 1 {
```

Comment:
```suggestion
    if signing_certs.is_empty() {
```

Reply:
Thanks, fixed in 2f4b690. I don't know why clippy warning aren't displayed in this workspace, they work elsewhere.